### PR TITLE
[feat][admin] Add offload managedLedgerOffloadThreshold RestAPI and CLI tools

### PR DESCRIPTION
**Pull request by tjiuming**
### Motivation

Add offload managedLedgerOffloadThreshold RestAPI and CLI tools

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [x] The REST endpoints
- [x] The admin CLI options
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/tjiuming/pulsar/pull/8

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java [hunk 1]
```diff
@@ -1965,6 +1965,32 @@ protected void internalSetOffloadThreshold(long newThreshold) {
         }
     }
 
+    protected void internalSetOffloadThresholdInSeconds(long newThreshold) {
```
Comment (Technoboy-) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java:1968]: We should make this method to be async.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java [hunk 2]
```diff
@@ -1965,6 +1965,36 @@ protected void internalSetOffloadThreshold(long newThreshold) {
         }
     }
 
+    protected CompletableFuture<Void> internalSetOffloadThresholdInSecondsAsync(long newThreshold) {
+        validateNamespacePolicyOperation(namespaceName, PolicyName.OFFLOAD, PolicyOperation.WRITE);
+        validatePoliciesReadOnlyAccess();
```
Comment (nodece) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java:1970]: Please switch to async.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java [hunk 3]
```diff
@@ -1965,6 +1965,36 @@ protected void internalSetOffloadThreshold(long newThreshold) {
         }
     }
 
+    protected CompletableFuture<Void> internalSetOffloadThresholdInSecondsAsync(long newThreshold) {
+        validateNamespacePolicyOperation(namespaceName, PolicyName.OFFLOAD, PolicyOperation.WRITE);
```
Comment (nodece) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java:1969]: Please switch to async.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java [hunk 4]
```diff
@@ -2088,6 +2088,57 @@ public void setOffloadThreshold(@PathParam("tenant") String tenant,
...
+            @ApiResponse(code = 404, message = "Namespace doesn't exist") })
+    public void getOffloadThresholdInSeconds(
+            @Suspended final AsyncResponse asyncResponse,
+            @PathParam("tenant") String tenant,
+            @PathParam("namespace") String namespace) {
+        validateNamespaceName(tenant, namespace);
+        validateNamespacePolicyOperationAsync(namespaceName, PolicyName.OFFLOAD, PolicyOperation.READ)
+                .thenCompose(__ -> getNamespacePoliciesAsync(namespaceName))
+                .thenAccept(policies -> {
+                    if (policies.offload_policies == null) {
```
Comment (nodece) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java:2106]:
```suggestion
                    if (policies.offload_policies != null) {
```

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiOffloadTest.java [hunk 5]
```diff
@@ -267,6 +275,97 @@ public void testOffloadPoliciesAppliedApi() throws Exception {
...
+
+        OffloadPoliciesImpl policies = new OffloadPoliciesImpl();
+        policies.setManagedLedgerOffloadThresholdInBytes(200L);
+        policies.setManagedLedgerOffloadThresholdInSeconds(200L);
+        policies.setManagedLedgerOffloadDeletionLagInMillis(400L);
+        policies.setManagedLedgerOffloadDriver("s3");
+        policies.setManagedLedgerOffloadBucket("buck2");
+
+        admin.namespaces().setOffloadThresholdInSeconds(myNamespace, 300);
+        assertEquals(300, admin.namespaces().getOffloadThresholdInSeconds(myNamespace));
```
Comment (nodece) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiOffloadTest.java:292]:
```suggestion
        assertEquals(admin.namespaces().getOffloadThresholdInSeconds(myNamespace), 300);
```

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiOffloadTest.java [hunk 6]
```diff
@@ -267,6 +275,97 @@ public void testOffloadPoliciesAppliedApi() throws Exception {
...
+
+        admin.namespaces().setOffloadPolicies(myNamespace, policies);
+        assertEquals(policies, admin.namespaces().getOffloadPolicies(myNamespace));
```
Comment (nodece) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiOffloadTest.java:295]:
```suggestion
        assertEquals(admin.namespaces().getOffloadPolicies(myNamespace), policies);
```

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiOffloadTest.java [hunk 7]
```diff
@@ -267,6 +275,97 @@ public void testOffloadPoliciesAppliedApi() throws Exception {
...
+
+        OffloadPoliciesImpl namespacePolicies = new OffloadPoliciesImpl();
+        namespacePolicies.setManagedLedgerOffloadThresholdInBytes(200L);
+        namespacePolicies.setManagedLedgerOffloadThresholdInSeconds(200L);
+        namespacePolicies.setManagedLedgerOffloadDeletionLagInMillis(400L);
+        namespacePolicies.setManagedLedgerOffloadDriver("s3");
+        namespacePolicies.setManagedLedgerOffloadBucket("buck2");
+
+        admin.namespaces().setOffloadThresholdInSeconds(myNamespace, 300);
+        assertEquals(300, admin.namespaces().getOffloadThresholdInSeconds(myNamespace));
```
Comment (nodece) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiOffloadTest.java:329]:
```suggestion
        assertEquals(admin.namespaces().getOffloadThresholdInSeconds(myNamespace), 300);
```

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiOffloadTest.java [hunk 8]
```diff
@@ -267,6 +275,97 @@ public void testOffloadPoliciesAppliedApi() throws Exception {
...
+
+        admin.namespaces().setOffloadPolicies(myNamespace, namespacePolicies);
+        assertEquals(namespacePolicies, admin.namespaces().getOffloadPolicies(myNamespace));
```
Comment (nodece) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiOffloadTest.java:332]:
```suggestion
        assertEquals(admin.namespaces().getOffloadPolicies(myNamespace), namespacePolicies);
```

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/NamespacesTest.java [hunk 9]
```diff
@@ -1398,15 +1398,18 @@ public void testSetOffloadThreshold() throws Exception {
...
         // set a default
         pulsar.getConfiguration().setManagedLedgerOffloadAutoTriggerSizeThresholdBytes(1);
+        pulsar.getConfiguration().setManagedLedgerOffloadThresholdInSeconds(100);
         // create the namespace
         admin.namespaces().createNamespace(namespace, Set.of(testLocalCluster));
         admin.topics().createNonPartitionedTopic(topicName.toString());
 
         admin.namespaces().setOffloadDeleteLag(namespace, 10000, TimeUnit.SECONDS);
         assertEquals(-1, admin.namespaces().getOffloadThreshold(namespace));
+        assertEquals(-1, admin.namespaces().getOffloadThresholdInSeconds(namespace));
```
Comment (nodece) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/NamespacesTest.java:1408]:
```suggestion
        assertEquals(admin.namespaces().getOffloadThresholdInSeconds(namespace), -1);
```

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/NamespacesTest.java [hunk 10]
```diff
@@ -1398,15 +1398,18 @@ public void testSetOffloadThreshold() throws Exception {
...
 
         // assert we get the default which indicates it will fall back to default
         assertEquals(-1, admin.namespaces().getOffloadThreshold(namespace));
+        assertEquals(-1, admin.namespaces().getOffloadThresholdInSeconds(namespace));
```
Comment (nodece) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/NamespacesTest.java:1412]:
```suggestion
        assertEquals(admin.namespaces().getOffloadThresholdInSeconds(namespace), -1);
```

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/NamespacesTest.java [hunk 11]
```diff
@@ -1415,15 +1418,21 @@ public void testSetOffloadThreshold() throws Exception {
...
                 Long.valueOf(-1));
+        assertEquals(ledgerConf.getLedgerOffloader().getOffloadPolicies().getManagedLedgerOffloadThresholdInSeconds(),
+                Long.valueOf(-1));
+
 
         // set an override for the namespace
         admin.namespaces().setOffloadThreshold(namespace, 100);
+        admin.namespaces().setOffloadThresholdInSeconds(namespace, 100);
         assertEquals(100, admin.namespaces().getOffloadThreshold(namespace));
+        assertEquals(100, admin.namespaces().getOffloadThresholdInSeconds(namespace));
```
Comment (nodece) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/NamespacesTest.java:1435]:
```suggestion
        assertEquals(admin.namespaces().getOffloadThresholdInSeconds(namespace), 100);
```

File: pulsar-client-tools-test/src/test/java/org/apache/pulsar/admin/cli/PulsarAdminToolTest.java [hunk 12]
```diff
@@ -804,11 +804,11 @@ public void namespaces() throws Exception {
         verify(mockNamespaces).clearOffloadDeleteLag("myprop/clust/ns1");
 
         namespaces.run(split(
-                "set-offload-policies myprop/clust/ns1 -r test-region -d aws-s3 -b test-bucket -e http://test.endpoint -mbs 32M -rbs 5M -oat 10M -oae 10s -orp tiered-storage-first"));
+                "set-offload-policies myprop/clust/ns1 -r test-region -d aws-s3 -b test-bucket -e http://test.endpoint -mbs 32M -rbs 5M -oat 10M -aots 100 -oae 10s -orp tiered-storage-first"));
```
Comment (nodece) [inline pulsar-client-tools-test/src/test/java/org/apache/pulsar/admin/cli/PulsarAdminToolTest.java:807]: Should be `-ts 100`?

File: pulsar-client-tools-test/src/test/java/org/apache/pulsar/admin/cli/PulsarAdminToolTest.java [hunk 13]
```diff
@@ -1414,11 +1421,11 @@ public void topicPolicies() throws Exception {
         verify(mockGlobalTopicsPolicies).removeOffloadPolicies("persistent://myprop/clust/ns1/ds1");
 
         cmdTopics.run(split("set-offload-policies persistent://myprop/clust/ns1/ds1 -d s3 -r" +
-                " region -b bucket -e endpoint -m 8 -rb 9 -t 10 -orp tiered-storage-first -g"));
+                " region -b bucket -e endpoint -m 8 -rb 9 -t 10 -oats 100 -orp tiered-storage-first -g"));
```
Comment (nodece) [inline pulsar-client-tools-test/src/test/java/org/apache/pulsar/admin/cli/PulsarAdminToolTest.java:1424]: Should be `-ts 100`?

File: pulsar-client-tools-test/src/test/java/org/apache/pulsar/admin/cli/PulsarAdminToolTest.java [hunk 14]
```diff
@@ -1701,10 +1708,10 @@ public void topics() throws Exception {
         cmdTopics.run(split("remove-delayed-delivery persistent://myprop/clust/ns1/ds1"));
         verify(mockTopics).removeDelayedDeliveryPolicy("persistent://myprop/clust/ns1/ds1") ;
 
-        cmdTopics.run(split("set-offload-policies persistent://myprop/clust/ns1/ds1 -d s3 -r region -b bucket -e endpoint -m 8 -rb 9 -t 10 -orp tiered-storage-first"));
+        cmdTopics.run(split("set-offload-policies persistent://myprop/clust/ns1/ds1 -d s3 -r region -b bucket -e endpoint -oats 50 -m 8 -rb 9 -t 10 -orp tiered-storage-first"));
```
Comment (nodece) [inline pulsar-client-tools-test/src/test/java/org/apache/pulsar/admin/cli/PulsarAdminToolTest.java:1711]: Should be `-ts 50`?

**Review by nodece — APPROVED** [review]

**Review by Technoboy- — APPROVED** [review]

**Pull request merged**